spec/models/spree/feed_product_spec.rb

```diff
@@ -1,55 +1,248 @@
+                   property: brand_property, value: 'Pamasaonic'}
+
+    describe "brand?" do
+      subject { feed_product.send :brand? }
+
+      it { is_expected.to be_truthy }
+    end
+
+    describe "#identifier_exists" do
+      subject { feed_product.send :identifier_exists }
+
+      it { is_expected.to eq 'no' }
+
+      context "when gtin exists" do
+        let(:gtin_property) { create :property, name: 'gtin',
+                              presentation: 'GTIN' }
+        let!(:gtin) { create :product_property, product: product,
+                      property: gtin_property,
+                      value: 123 }
+
+        it { is_expected.to eq 'yes' }
+
+        describe "#gtin?" do
+          subject { feed_product.send :gtin? }
+
+          it { is_expected.to be_truthy }
+        end
+      end
+
+      context "when mpn exists" do
+        let(:mpn_property) { create :property, name: 'mpn',
+                              presentation: 'MPN' }
+        let!(:mpn) { create :product_property, product: product,
+                      property: mpn_property,
+                      value: 123 }
+
+        it { is_expected.to eq 'yes' }
+
+        describe "#mpn?" do
+          subject { feed_product.send :mpn? }
+
+          it { is_expected.to be_truthy }
+        end
+      end
+    end
+  end
+
+  context "when brand property does not exist" do
+    describe "#identifier_exists" do
+      subject { feed_product.send :identifier_exists }
+
+      it { is_expected.to eq 'no' }
+
+      context "when gtin exists" do
+        let(:gtin_property) { create :property, name: 'gtin',
+                              presentation: 'GTIN' }
+        let!(:gtin) { create :product_property, product: product,
+                      property: gtin_property,
+                      value: "GTIN123" }
+
+        it { is_expected.to eq 'no' }
+
+        describe "#gtin" do
+          subject { feed_product.send :gtin }
+          it { is_expected.to eq "GTIN123" }
+        end
+      end
+
+      context "when mpn exists" do
+        let(:mpn_property) { create :property, name: 'mpn',
+                              presentation: 'MPN' }
+        let!(:mpn) { create :product_property, product: product,
+                      property: mpn_property,
+                      value: "MPN123" }
+
+        it { is_expected.to eq 'no' }
+
+        describe "#mpn" do
+          subject { feed_product.send :mpn }
+          it { is_expected.to eq "MPN123" }
+        end
+      end
+    end
+  end
+
+  describe "#tax_rate" do
+    subject { feed_product.send :tax_rate }
+
+    let!(:tax_rate_1) { create :tax_rate, tax_category: product.tax_category }
+    let!(:tax_rate_2) { create :tax_rate, tax_category: product.tax_category,
+                       amount: 0.5 }
+    context "when there are tax rates on line items for this product" do
+      let(:order) { create :order }
+      let(:line_item_rate_1) { create :line_item, product: product,
```

Comment:
Can we use `do/end` for multiline blocks and align all arguments? This applies to many `let` blocks in this PR.